### ISSUE TITLE
Tensorflow 2.0 rc0 imports are not properly followed

### ISSUE THREAD

**Issue by pandrey-fr** [PyCQA/pylint#3080]
To be fair, I do not know if the issue comes from pylint, from tensorflow or from something wrong in my configuration, but here is an issue that appeared on both my systems (a Windows 10 and a Linux Mint one) after upgrading my main tensorflow installation to the recently-released `2.0.0-rc0` (my previous installation was the `2.0.0-beta1` one, which was linted just fined).

Basically, pylint does not lint inheritance properly. More precisely, tensorflow imports seem to be working, e.g. I can write a custom layer inheriting from `tf.keras.layers.Layer` without triggering a linter warning or error message, but Pylint does not handle inheritance properly; for instance non-overridden inherited public methods are not counted (resulting in `too-few-public-methods` warnings), non-overriden abstract methods do not trigger warnings, and overridden methods that do not comprise a `super` call (typically, abstract methods) are not checked as to parent-method signature respect (resulting in misplaced `unused-argument` warnings and lacking `arguments-differ` ones).

### Steps to reproduce

Here is a dummy example of code that will not be linted correctly with `2.0.0-rc0` installed.

```python

import tensorflow as tf

class ExampleLayer(tf.keras.layers.Layer):

    # should not trigger 'too-few-public-methods' as many are inherited
    # should trigger 'Method `compute_output_signature` is abstract in
    #                 class `tf.keras.layers.Layer` but is not overridden'

    def __init__(self, units):
        """Instantiate the layer, making it a LSTM with less arguments."""
        super().__init__()
        self.lstm = tf.keras.layers.LSTM(units)

    def call(self, inputs, mask):  # should trigger 'arguments-differ' warning
        return self.lstm(inputs, mask=mask)
```

### Current and Expected behavior

Apart from valid docstring-related warnings, and as indicated in in-code comments, the code above should:
* Not trigger a 'too-few-public-methods' warning, as many are inherited.
* Trigger 'Method `compute_output_signature` is abstract in class `tf.keras.layers.Layer` but is not overridden'
* Trigger an `arguments-differ` warning on the `call` method due to signature mismatch (which is in fact expected by the API, but it should be up to the user to disable the warning if relevant).

Note that the behavior I expect is the one that pylint (same version) used to have before I updated from `2.0.0-beta1` to `2.0.0-rc0`.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```

**Comment by pandrey-fr** [comment]
I totally understand that nobody has had time to reproduce / investigate this issue, but I would just like to state that it still active as of TensorFlow 2.0 and 2.1 official releases. Perhaps it is due to the way TF exports classes?

**Comment by pandrey-fr** [comment]
Update : linting seems to be running fine again on TensorFlow 2.2.0-rc0 (tested on a single Windows installation as of now).

**Comment by pandrey-fr** [comment]
I confirm that the bug is, as far as I am concerned (on multiple machines and OSes) resolved as of the release of TensorFlow 2.2.

**Comment by DanielNoord** [comment]
This sounds more like it is a bug on an older version. Are we sure this reproduces on `main`?

**Comment by Pierre-Sassoulas** [comment]
We could not reproduce at the time. Could you gives some details @stadawi12 ? How similar is it ?

**Comment by stadawi12** [comment]
After a bit of digging around I have found the solution. What I needed to do was to ensure that I am running the correct Python version, as I am working in a python environment and also I needed to read in the .pylintrc file from my working directory, I did that by configuring the coc-settings.json file.
```
{
`"python.pythonPath":` "/home/dawid/.config/.venv/bin/python",
"python.linting.pylintArgs": ["--rcfile", "/home/dawid/projects/machine_learning/lstm/.pylintrc"],
}
```
Now pylint no longer throws an error when I use modules from torch, although I am still not sure why pylint has issues with seeing torch modules.
Adding the above has solved the issue as well as updating coc to the latest version, I am very new to all this and I got really lost, sorry for the trouble.
